### ISSUE TITLE
Cannot use with Rack 3.0 because of Sinatra (resque-web); Blocked up grade to Rails 7.1

### ISSUE THREAD

**Issue by danlo** [resque/resque#1890]
TL;DR:  Rails 7.1 requires Rack 3 and Resque-web cannot support Rack 3 (due to sinatra).  Can Resque-web be made into it's own gem?

Greetings,

This appears to be a partial repeat of #1492.

Sinatra cannot use Rack 3, due to the issues laid out (and in progress) here.    
 https://github.com/sinatra/sinatra/issues/1797#issuecomment-1452822705 

> Just like to point out that [rack-session-2.0.0](https://rubygems.org/gems/rack-session) now requires rack >= 3.0.0, however [sinatra-3.0.5](https://rubygems.org/gems/sinatra/versions/3.0.5) still requires rack ~> 2.2, which makes the rack-session-2.0.0 gem incompatible with sinatra <= 3.0.5. This will probably start biting developers.

Which for rails 7.1 looks like this: 

    rails (~> 7.1) was resolved to 7.1.2, which depends on
      actionpack (= 7.1.2) was resolved to 7.1.2, which depends on
        rack-session (>= 1.0.1) was resolved to 2.0.0, which depends on
          rack (>= 3.0.0)

Since Sinatra cannot run rack 3; Rails 7.1 and Resque are incompatible. 

@dentarg states that:   https://github.com/sinatra/sinatra/issues/1797#issuecomment-1826754094
> Not yet, merging to main will mark the end for Sinatra 3.x (at least I'm not interested in maintaining multiple branches). I just created https://github.com/sinatra/sinatra/issues/1962 with things that should be released for 3.x.

Which suggests that Sinatra 4.x may be a while.   

In the issue #1492, it was suggested that resque-web be moved into it's own separate gem  

https://github.com/resque/resque/issues/1492#issuecomment-301284335 

---

Given, that Sinatra 4.x may be a while, could there be a release of resque w/o resque-web, so that those of us who use Resque could continue doing so on Rails 7.1?

-daniel

**Comment by PatrickTulskie** [comment]
Might not be a bad idea. We are about to embark on this upgrade with one app too and I think we are going to have the same problem. Going to start looking into solutions. I like the idea of splitting off resque-web into its own gem, but I want to make sure it still has the ease of use and compatibility with the rest of the plugins in the ecosystem.

**Comment by varyonic** [comment]
Really? We're using Rails 7.1 with Rack 2.2 just fine.

**Comment by danlo** [comment]
> Really? We're using Rails 7.1 with Rack 2.2 just fine.

Thank you for the confirmation that it works.  Looking though the dependencies I found that this setup works
```
  gem 'rails', '~> 7.1'
  gem 'rack-session', '~> 1.0', '>= 1.0.2'
```

**Comment by dentarg** [comment]
Sinatra 4 has been out for a while FYI

**Comment by JDrizzy** [comment]
> Sinatra 4 has been out for a while FYI

Upgrading sinatra to version 4 appears to resolve this issue as well. I'm still testing the rails upgrade just to make sure.

`bundle update sinatra`

**Comment by mishina2228** [comment]
@danlo As mentioned at https://github.com/resque/resque/pull/1891#issuecomment-2117450936 ,  [Sinatra 4.0](https://github.com/sinatra/sinatra/blob/main/CHANGELOG.md#400--2024-01-19) supports Rack 3.0, so we can use rack-session 2.0 with Rails 7.1.
